code/parse/sexp.cpp

```diff
@@ -19319,54 +19231,14 @@ void sexp_primitive_sensors_set_range(int n)
 // AutoNav/AutoPilot system SEXPS
 
 //text: set-nav-carry
-//args: 1+, Ship/Wing name
-void set_nav_carry_status(int node)
-{
-	int n=node, i;
-	char *name;
-	bool skip;
-
-	while (n != -1)
-	{
-		name = CTEXT(n);
-		skip = false;
-
-		for (i = 0; i < MAX_WINGS; i++)
-		{
-			if (!stricmp(Wings[i].name, name))
-			{
-                Wings[i].flags.set(Ship::Wing_Flags::Nav_carry);
-				skip = true;
-				break;
-			}
-		}
-
-		if (!skip)
-		{
-			for (i = 0; i < MAX_SHIPS; i++)
-			{
-				if (Ships[i].objnum != -1 && !stricmp(Ships[i].ship_name, name))
-				{
-                    Ships[i].flags.set(Ship::Ship_Flags::Navpoint_carry);
-					break;
-				}
-			}
-		}
-		
-		// move to next ship/wing in list
-		n = CDR(n);
-	}
-}
-
 //text: unset-nav-carry
 //args: 1+, Ship/Wing name
-void unset_nav_carry_status(int node)
+void set_unset_nav_carry_status(int node, bool set_it)
```

Comment:
what do you think about adding a comment which explains the parameter set_it.
Its not really clear in the signature.

Reply:
The parameter is self-explanatory.  That sort of function signature is quite clear, and used in numerous places in the code.

Reply:
Think of people like me, who do not have the years of experience in the codebase you have :)

Reply:
Sure.  But what you do is a) look at the function definition, b) look at similar function definitions, and c) look at the body of the function.

Any one of those three can explain what's happening, but in this case, a) is sufficient.  The name of the function indicates that something is being set or unset, and the boolean flag (which can only take two values) indicates whether the thing is being set.